pkg/parser/node/accessor.go

```diff
@@ -68,3 +62,24 @@ func (node *Node) GetSliceOfNonEmptyStrings() ([]string, error) {
 		return nil, fmt.Errorf("%w: field should be a string or a list of values", parsererror.ErrParsing)
 	}
 }
+
+func expandEnvironmentVariables(s string, env map[string]string) string {
+	const maxExpansionIterations = 10
+
+	for i := 0; i < maxExpansionIterations; i++ {
+		beforeExpansion := s
+
+		for key, value := range env {
```

Comment:
This is not deterministic and can lead to weird cases when you have two environment varaibales where one is a prefix of another. For example, `CIRRUS=true` and `CIRRUS_BRANCH=main`. Cirrus cloud sorts the environment variables from longest tot shortest before evaluating.

Reply:
Thanks for the nit, fixed in https://github.com/cirruslabs/cirrus-cli/pull/69/commits/486ef08a80f0747532d9588433786011b9e95d19.